src/plone/restapi/services/content/tus.py

```diff
@@ -394,3 +394,14 @@ def expires(self):
         else:
             expiration = time.time() + self.expiration_period
         return formatdate(expiration, False, True)
+
+
+@implementer(IStorage)
+class TUSUploadStorable:
+    """Special storage to skip reading and writing of the upload since it's already on disk"""
+
+    def store(self, data, blob):
+        if not isinstance(data, TUSUpload):
+            raise NotStorable('Could not store data (not of "FileUpload").')
```

Comment:
```suggestion
            raise NotStorable('Could not store data (not of type "TUSUpload").')
```

Reply:
@djay Now that I'm looking at things more carefully, I'm wondering if this special storage adapter is really needed.

Previously, an open file was passed into the NamedBlobFile constructor. This should end up using the [FileDescriptorStorable](https://github.com/plone/plone.namedfile/blob/master/plone/namedfile/storages.py#L62) which uses [blob.consumeFile](https://github.com/zopefoundation/ZODB/blob/891a8a6e056e58eedb010f8971a492a4b083f2e4/src/ZODB/blob.py#L227), which is already using rename_or_copy_blob.